runtime/src/accounts_db.rs

```diff
@@ -746,6 +766,10 @@ impl AccountsDB {
     }
 
     pub fn hash_account_data(fork: Fork, lamports: u64, data: &[u8], pubkey: &Pubkey) -> Hash {
+        if lamports == 0 {
```

Comment:
this one's interesting, how come?

Reply:
Because if we purge, we would not maintain the correct bank hash value.

Reply:
ah!